data-pipeline-ffi/src/trace_exporter.rs

```diff
@@ -1,59 +1,101 @@
 // Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
 // SPDX-License-Identifier: Apache-2.0
 
-use data_pipeline::trace_exporter::TraceExporter;
-use data_pipeline::trace_exporter::TraceExporterBuilder;
+use data_pipeline::trace_exporter::{
+    ResponseCallback, TraceExporter, TraceExporterInputFormat, TraceExporterOutputFormat,
+};
 use ddcommon_ffi::{
     slice::{AsBytes, ByteSlice},
-    CharSlice,
+    CharSlice, MaybeError,
 };
-use std::ffi::c_char;
+use std::{ffi::c_char, ptr::NonNull};
 
+/// Create a new TraceExporter instance.
+///
+/// # Arguments
+///
+/// * `out_handle` - The handle to write the TraceExporter instance in.
+/// * `url` - The URL of the Datadog Agent to communicate with.
+/// * `tracer_version` - The version of the client library.
+/// * `language` - The language of the client library.
+/// * `language_version` - The version of the language of the client library.
+/// * `language_interpreter` - The interpreter of the language of the client library.
+/// * `input_format` - The input format of the traces. Setting this to Proxy will send the trace
+///   data to the Datadog Agent as is.
+/// * `output_format` - The output format of the traces to send to the Datadog Agent. If using the
+///   Proxy input format, this should be set to format if the trace data that will be passed through
+///   as is.
+/// * `agent_response_callback` - The callback into the client library that the TraceExporter uses
+///   for updated Agent JSON responses.
 #[no_mangle]
 pub unsafe extern "C" fn ddog_trace_exporter_new(
-    host: CharSlice,
-    port: u16,
+    out_handle: NonNull<Box<TraceExporter>>,
+    url: CharSlice,
     tracer_version: CharSlice,
     language: CharSlice,
     language_version: CharSlice,
     language_interpreter: CharSlice,
-) -> *mut TraceExporter {
-    let builder = TraceExporterBuilder::default();
-
-    let exporter = builder
-        .set_host(host.to_utf8_lossy().as_ref())
-        .set_port(port)
+    input_format: TraceExporterInputFormat,
+    output_format: TraceExporterOutputFormat,
+    agent_response_callback: extern "C" fn(*const c_char),
+) -> MaybeError {
+    let callback_wrapper = ResponseCallbackWrapper {
+        response_callback: agent_response_callback,
+    };
+    // TODO - handle errors
```

Comment:
By errors you mean logs? Or should we add a TODO: add logs callback as well
?

Reply:
This is about returning errors to the caller of the function. I intentionally left logging out of this API for now.